### PR TITLE
88621024 electric load center storage battery crash

### PR DESCRIPTION
**Pull request by JasonGlazer**
Addresses #4595 and includes test case. A counter was added to the loop so that it could not iterate indefinitely. The loop was placed in a new routine so that unit tests could exercise it.

### CODE REVIEW

**Comment by JasonGlazer** [comment]
@Myoldmopar @mjwitte @lefticus How can I figure out if the CI system has gotten stuck? The Win64-Windows-7-VisualStudio-12 has said "Build Pending" while all of the other tests were completed and pass for at least half a day. Should I try to "delete the results and force a rebuild" or would that not work because build is still pending anyway?

**Comment by mjwitte** [comment]
It's hard to know.  There have been other reports from Win64 for commits 
that came in after yours, so I'd say go ahead and delete/rebuild.

**Comment by Myoldmopar** [comment]
I concur; go ahead and delete if you haven't already.

**Comment by JasonGlazer** [comment]
Looks like it reran and everything is ok now.

File: src/EnergyPlus/ManageElectricPower.cc [hunk 1]
```diff
@@ -3192,6 +3175,82 @@ namespace ManageElectricPower {
...
+		//       AUTHOR         B. Griffith
+		//       DATE WRITTEN   June-August 2008
+		//       MODIFIED       BG May 2009, added EMS
+		//                      BN (FSEC) Feb 2010 (pass out two storage values)
+		//                      Y. KyungTae & W. Wang July-August, 2011 Added a battery model
+		//       RE-ENGINEERED  Jason Glazer, GARD Analytics, February 2015, refactor charge calculation into a function
+
+		// PURPOSE OF THIS FUNCTION:
+		// Calculate the current for battery discharge in a separate function so that it could be called from the unit tests
+
```
Comment (jasondegraw) [inline src/EnergyPlus/ManageElectricPower.cc]: @JasonGlazer I added this function information block - maybe it isn't necessary but I think it is helpful. Does it look OK to you?

Comment (JasonGlazer) [inline src/EnergyPlus/ManageElectricPower.cc]: Looks good thanks.

**Comment by JasonGlazer** [comment]
@jasondegraw Maybe this just needs to be merged again.

**Comment by JasonGlazer** [comment]
@jasondegraw I tried to merge to see if the tests were failing because this branch was not close enough to develop. It required a manual change to resolve an merge conflict. That was the first time I needed to do that and maybe I did it wrong because now the unit tests do not pass related to a problem in unrelated code when the SizePurchasedAirTest.Test1 (in PurchasedAirManager.unit.cc) was running and trying to access the UnitarySysEqSizing array for an index that does not exist when executing the line:

 	SizePurchasedAir( PurchAirNum );

I did not push this back into the repo since it did not pass the unit tests but I am not sure what to do. Advice?

**Comment by jasondegraw** [comment]
@JasonGlazer I encountered the same issue yesterday. It looks like the order that the test are run in matters, which is not great - but there's not much that can be done to change that right now. I added the allocation of UnitarySysEqSizing in the test and that fixed the crash for me, but I didn't push to see what would happen. I can push that when I get into the office, or you can push your fix (just make sure to add deallocations of the globals).

**Comment by Myoldmopar** [comment]
@jasondegraw how about you go ahead and push your fix when you get into the
office. It's possibly affecting everyone, so the quicker it gets into
develop the better.

**Comment by jasondegraw** [comment]
@JasonGlazer @Myoldmopar Well, just so everyone knows, I misremembered which branch I had this on. It was actually on the FluidCooler branch. I've pushed that, so I guess we wait and see what happens with CI

**Pull request merged**